cmd/sops/decrypt.go

```diff
@@ -45,6 +49,9 @@ func decrypt(opts decryptOpts) (decryptedFile []byte, err error) {
 		return extract(tree, opts.Extract, opts.OutputStore)
 	}
 	decryptedFile, err = opts.OutputStore.EmitPlainFile(tree.Branches)
+	if bsepe, ok := err.(json.BinaryStoreEmitPlainError); ok {
+		err = fmt.Errorf("%s. %s", bsepe.Error(), notBinaryHint)
+	}
```

Comment:
Why not use `errors.Is(err, json.BinaryStoreEmitPlainError)`, and define the latter as , for example
```
var BinaryStoreEmitPlainError = errors.New("error emitting binary store")
```
And when you throw it, you do ,for example
```
return nil, fmt.Errorf("%w: 'data' key in tree does not have a string value", BinaryStoreEmitPlainError)
```
Then you don't need a custom error-type

Reply:
The why answer is easy: because I've never seen that before :) This looks a lot nicer, I've switched to it in 51956d9034498632957cc878c3ef8205f68b58a5.